src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+
 public class WorkWithFile {
+    static final String SUPPLY = "supply";
+    static final String BUY = "buy";
+    static final String RESULT = "result";
+    static final String SEPARATOR = ",";
+    static final String SPLITTER = " ";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToNewFile(toFileName, getReport(readFromFile(fromFileName)));
+    }
+
+    public String getReport(String dataFromFile) {
+        String[] data = dataFromFile.split(SPLITTER);
+        int supplyCount = 0;
+        int buyCount = 0;
+        for (String value : data) {
+            int splitIndex = value.indexOf(SEPARATOR);
+            int count = Integer.parseInt(value.substring(splitIndex + 1));
+            if (value.substring(0, splitIndex).equals(SUPPLY)) {
+                supplyCount += count;
+            } else if (value.substring(0, splitIndex).equals(BUY)) {
+                buyCount += count;
+            }
```

Comment:
[try to solve using split here instead of indexOf and substring](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#use-split-instead-of-substring)
also, don't forget to create constants for array indexes